### PR TITLE
Add sphinx setup for build_wheel to get docstrings, enable OpenMP

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: packaging/python_wheel/docker_build_wheel.sh [hunk 1]
```diff
@@ -77,6 +77,21 @@ build_boost()
         install
 }
 
+# Needed for docstrings generation and system doxygen is too old
```
Comment (mgeplf) [inline packaging/python_wheel/docker_build_wheel.sh]: sigh

File: packaging/python_wheel/docker_build_wheel.sh [hunk 2]
```diff
@@ -77,6 +77,21 @@ build_boost()
...
+{
+    [[ -e /usr/local/bin/doxygen ]] && return
+
+    # after 1.8.6, we can't use anymore system flex
+    # https://stackoverflow.com/questions/20844635
+    tar xfz $EXT_SRC/Release_1_8_5.tar.gz -C /tmp
+    cd /tmp/doxygen-Release_1_8_5
+
+    ./configure
+    make -j8
```
Comment (mgeplf) [inline packaging/python_wheel/docker_build_wheel.sh]: do in one make invocation?

File: packaging/python_wheel/docker_build_wheel.sh [hunk 3]
```diff
@@ -90,6 +105,12 @@ build_brain()
     build_boost $PYTHON $PYTHON_INC
 
     /opt/python/$version/bin/pip install "numpy==$NUMPY_VERSION"
+    /opt/python/$version/bin/pip install "sphinx==1.3.6" lxml
+
+    # make sphinx-build visible for FindSphinx.cmake
```
Comment (mgeplf) [inline packaging/python_wheel/docker_build_wheel.sh]: this is just going to turn into hacks on top of hacks :)

File: packaging/python_wheel/docker_build_wheel.sh (hunk 2 above)
Comment (tribal-tec) [inline packaging/python_wheel/docker_build_wheel.sh]: tried. does not work of course...

File: packaging/python_wheel/docker_build_wheel.sh (hunk 3 above)
Comment (tribal-tec) [inline packaging/python_wheel/docker_build_wheel.sh]: that's what software engineering is all about :p

File: packaging/python_wheel/docker_build_wheel.sh [hunk 4]
```diff
@@ -77,6 +77,21 @@ build_boost()
         install
 }
 
+# Needed for docstrings generation and system doxygen is too old
+build_doxygen()
+{
+    [[ -e /usr/local/bin/doxygen ]] && return
+
+    # after 1.8.6, we can't use anymore system flex
```
Comment (hernando) [inline packaging/python_wheel/docker_build_wheel.sh]: Compile flex then :P

File: packaging/python_wheel/docker_build_wheel.sh (hunk 3 above)
Comment (hernando) [inline packaging/python_wheel/docker_build_wheel.sh]: Can't we fix FindSphinx.cmake or provide CMake a hint with the location instead?

Comment (tribal-tec) [inline packaging/python_wheel/docker_build_wheel.sh]: https://github.com/BlueBrain/Pydoxine/pull/11

**Pull request merged**